### PR TITLE
tree: rename checkout to branch in API

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Alings the shared tree API with the names used in shared tree core, which are the new "branch" related names.

## Breaking Changes

This is all breaking API renames, but in an internal package.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: api-report/tree.api.md [hunk 1]
```diff
@@ -734,16 +734,16 @@ export type IsEvent<Event> = Event extends (...args: any[]) => any ? true : fals
...
 // @alpha
-export interface ISharedTreeCheckout extends AnchorLocator {
+export interface ISharedTreeBranch extends AnchorLocator {
     readonly context: EditableTreeContext;
     readonly editor: IDefaultEditBuilder;
-    readonly events: ISubscribable<CheckoutEvents>;
+    readonly events: ISubscribable<BranchEvents>;
     readonly forest: IForestSubscription;
-    fork(): ISharedTreeCheckoutFork;
+    fork(): ISharedTreeFork;
```
Comment (jenn-le) [inline api-report/tree.api.md:746]: I feel like there has to be some tree-related terminology for fork. Stem? Shoot? 🤔

File: packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts [hunk 2]
```diff
@@ -2083,14 +2083,14 @@ const testValueSchema = namedTreeSchema({
  * Inserts a single node under the root of the tree with the given value.
  * Use {@link peekTestValue} to read the value.
  */
-function pushTestValue(checkout: ISharedTreeCheckout, value: TreeValue): void {
+function pushTestValue(checkout: ISharedTreeBranch, value: TreeValue): void {
 	insert(checkout, 0, value);
 }
```
Comment (jenn-le) [inline packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts:2088]:
```suggestion
function pushTestValue(branch: ISharedTreeBranch, value: TreeValue): void {
	insert(branch, 0, value);
}
```

Comment (jenn-le) [reply packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts:2088]: Or, tree to be consistent with the functions below:

```suggestion
function pushTestValue(tree: ISharedTreeBranch, value: TreeValue): void {
	insert(tree, 0, value);
}
```

**Review by jenn-le — APPROVED** [review]

**Comment by noencke** [comment]
@CraigMacomber @jenn-le 

What was the motivation for this change? Do we not like the term "checkout"? I find the new state of things to be more confusing. The scheme before:

* Branch: the underlying primitive for creating branches in the edit/commit graph. Not part of public API.
* Checkout: a view of the SharedTree for users to do stuff with.
* CheckoutFork: a checkout which is not the root checkout; it has been forked off of a parent checkout

After this rename, "Checkout" becomes "Branch". The name "branch" seems to imply there is some significance to the fact that it is a branch, but the common-case simple app (which never calls `fork()`) probably won't care about branches at all, so it'd be nice if they never had to see the name "branch". And then of course, it is confusing to have "Fork" which is a "Branch", but not the root branch. "Fork" and "Branch" now are both in the public API vocabulary but have subtly different meanings.

"Checkout" is a name that was selected for historical reasons above anything else, so I'm not attached to it, but I still think it was better than "Branch". What do you think about calling it "View" instead? I think having `SharedTreeView` be the interface for interacting with the tree, and having `SharedTreeFork` extend `SharedTreeView` makes sense and would be intuitive enough to users. It would also mean that the distinction between "Branch" and "Fork" is now only relevant to implementers, and is much less subtle.

**Comment by noencke** [comment]
> @CraigMacomber @jenn-le
> 
> What was the motivation for this change? Do we not like the term "checkout"? I find the new state of things to be more confusing. The scheme before:
> 
> * Branch: the underlying primitive for creating branches in the edit/commit graph. Not part of public API.
> * Checkout: a view of the SharedTree for users to do stuff with.
> * CheckoutFork: a checkout which is not the root checkout; it has been forked off of a parent checkout
> 
> After this rename, "Checkout" becomes "Branch". The name "branch" seems to imply there is some significance to the fact that it is a branch, but the common-case simple app (which never calls `fork()`) probably won't care about branches at all, so it'd be nice if they never had to see the name "branch". And then of course, it is confusing to have "Fork" which is a "Branch", but not the root branch. "Fork" and "Branch" now are both in the public API vocabulary but have subtly different meanings.
> 
> "Checkout" is a name that was selected for historical reasons above anything else, so I'm not attached to it, but I still think it was better than "Branch". What do you think about calling it "View" instead? I think having `SharedTreeView` be the interface for interacting with the tree, and having `SharedTreeFork` extend `SharedTreeView` makes sense and would be intuitive enough to users. It would also mean that the distinction between "Branch" and "Fork" is now only relevant to implementers, and is much less subtle.

[Resolved in this PR](https://github.com/microsoft/FluidFramework/pull/14765), thank you!

**Pull request merged**